tests/createOption.test.ts

```diff
@@ -70,32 +69,44 @@ describe("Option Creation", () => {
         .signers([authorityKeypair])
         .rpc();
 
-      expect(optionCreatedEvent.poolAccount).to.eql(poolAccountKey);
-      expect(optionCreatedEvent.option).to.eql(optionTwoAccountKey);
-      expect(optionCreatedEvent.title).to.eql(optionTitle);
+      expect(optionCreatedEvent.poolAccount).toEqual(poolAccountKey);
+      expect(optionCreatedEvent.option).toEqual(optionTwoAccountKey);
+      expect(optionCreatedEvent.title).toEqual(optionTitle);
     } catch (e) {
-      expect(e.message).to.include("PoolOptionDoesNotMatchHash");
+      expect(e.message).toContain("PoolOptionDoesNotMatchHash");
     }
   });
 
-  it("does not allow a random wallet to create an option", async () => {
-    const randomKeypair = await generateKeypair();
-    const authorityKeypair = await getLocalAccount();
-    const title = "What was the nature of the Trump assassination?";
-    const optionTitle = "Lone Wolf";
-    const imageUrl = "https://example.com/image.png";
-    const description =
-      "This is a pool to guess the winner of the 2024 UEFA European Football Championship (Euro 2024).";
+  it("allows only the pool creator to create options", async () => {
+    const kp1 = await generateKeypair();
+    const kp2 = await generateKeypair();
+    const randomSuffix = Math.random().toString(36).substring(7);
     const { poolAccountKey } = await createPool(
-      title,
-      authorityKeypair,
-      imageUrl,
-      description,
+      `pool1_${randomSuffix}`,
+      kp1,
+      "",
+      "",
     );
+    const optionTitle = `option1_${randomSuffix}`;
+
+    // failure case
     try {
-      await createOption(optionTitle, randomKeypair, poolAccountKey);
+      await createOption(optionTitle, kp2, poolAccountKey);
+      throw new Error("This try block should have errored above");
     } catch (e) {
-      expect(e.message).to.include("An address constraint was violated");
+      expect(e.message).toContain("Pool account does not match derived key");
```

Comment:
same here, as a test improvement

Reply:
Have answered this in [another comment](https://github.com/Degen-Markets/degen-markets-solana/pull/9#discussion_r1796745972), pls see.

Since you highlighted these lines, it reminds me - Do you have any thoughts on using the error message (`Pool account does not match derived key` vs the error code (`PoolAccountDoesNotMatch`)? Or is either one fine?

Reply:
Either one is fine for me ser